server/svix-server/src/v1/endpoints/endpoint.rs

```diff
@@ -691,4 +711,366 @@ mod tests {
+        assert_eq!(
+            get_endpoint(&client, &app_1, &app_1_ep_1.id).await.unwrap(),
+            app_1_ep_1
+        );
+        assert_eq!(
+            get_endpoint(&client, &app_1, &app_1_ep_2.id).await.unwrap(),
+            app_1_ep_2
+        );
+        assert_eq!(
+            get_endpoint(&client, &app_2, &app_2_ep_1.id).await.unwrap(),
+            app_2_ep_1
+        );
+        assert_eq!(
+            get_endpoint(&client, &app_2, &app_2_ep_2.id).await.unwrap(),
+            app_2_ep_2
+        );
+
+        // Can't read from incorrect app
+        get_404(&client, &app_2, &app_1_ep_1.id).await.unwrap();
+        get_404(&client, &app_2, &app_1_ep_2.id).await.unwrap();
+        get_404(&client, &app_1, &app_2_ep_1.id).await.unwrap();
+        get_404(&client, &app_1, &app_2_ep_2.id).await.unwrap();
+
+        // UPDATE
+        let app_1_ep_1_id = app_1_ep_1.id;
+        let app_1_ep_1: EndpointOut = client
+            .put(
+                &format!("api/v1/app/{}/endpoint/{}/", app_1, app_1_ep_1_id),
+                endpoint_in(EP_URI_APP_1_EP_1_VER_2),
+                StatusCode::OK,
+            )
+            .await
+            .unwrap();
+        assert_eq!(app_1_ep_1.url, EP_URI_APP_1_EP_1_VER_2);
+
+        // CONFIRM UPDATE
+        assert_eq!(
+            get_endpoint(&client, &app_1, &app_1_ep_1_id).await.unwrap(),
+            app_1_ep_1
+        );
+
+        // LIST
+        let list_app_1: ListResponse<EndpointOut> = client
+            .get(&format!("api/v1/app/{}/endpoint/", &app_1), StatusCode::OK)
+            .await
+            .unwrap();
+        assert_eq!(list_app_1.data.len(), 2);
+        assert!(list_app_1.data.contains(&app_1_ep_1));
+        assert!(list_app_1.data.contains(&app_1_ep_2));
+
+        let list_app_2: ListResponse<EndpointOut> = client
+            .get(&format!("api/v1/app/{}/endpoint/", &app_2), StatusCode::OK)
+            .await
+            .unwrap();
+        assert_eq!(list_app_2.data.len(), 2);
+        assert!(list_app_2.data.contains(&app_2_ep_1));
+        assert!(list_app_2.data.contains(&app_2_ep_2));
+
+        // DELETE
+        delete_endpoint(&client, &app_1, &app_1_ep_1.id)
+            .await
+            .unwrap();
+        delete_endpoint(&client, &app_1, &app_1_ep_2.id)
+            .await
+            .unwrap();
+        delete_endpoint(&client, &app_2, &app_2_ep_1.id)
+            .await
+            .unwrap();
+        delete_endpoint(&client, &app_2, &app_2_ep_2.id)
+            .await
+            .unwrap();
+
+        // CONFIRM DELETION
+        get_404(&client, &app_1, &app_1_ep_1.id).await.unwrap();
+        get_404(&client, &app_1, &app_1_ep_2.id).await.unwrap();
+        get_404(&client, &app_2, &app_2_ep_1.id).await.unwrap();
+        get_404(&client, &app_2, &app_2_ep_2.id).await.unwrap();
+
+        delete_test_app(&client, app_1).await.unwrap();
+        delete_test_app(&client, app_2).await.unwrap();
```

Comment:
You don't need to manually cleanup (this applies to other places in the code too), because:
1. It's tedious and annoying to do.
2. We'll be doing it automatically in the near future anyway.
3. It's error-prone (you'll surely forget something, so automatic is better).
4. It doesn't add much.